src/Umbraco.Web/Editors/EntityController.cs

```diff
@@ -144,6 +144,37 @@ public IEnumerable<int> GetPath(int id, UmbracoEntityTypes type)
             return foundContent.Path.Split(new[] {','}, StringSplitOptions.RemoveEmptyEntries).Select(int.Parse);
         }
 
+
+
+
+        public dynamic GetUrl(int id, UmbracoEntityTypes type)

```

Comment:
Can we not return a `dynamic` C# object in an API response, perhaps see if we have an existing C# model/entity we can return or additionally as the code below is only returning a JSON object with a single property of `url` as a string.

Then perhaps this method should be refactored to simply use string as the return type & the calling AngularJS resource be modified to only return a string.

I assume you had some help here @madsrasmussen from @perploug perhaps?